presto-mysql/src/test/java/io/prestosql/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -219,6 +239,101 @@ public void testDecimalExceedingPrecisionMax()
         testUnsupportedDataType("decimal(50,0)");
     }
 
+    @DataProvider
+    public Object[][] testDecimalExceedingPrecisionMaxProvider()
+    {
+        return new Object[][] {
+                {60, 20},
+                {65, 25},
+        };
+    }
+
+    @Test(dataProvider = "testDecimalExceedingPrecisionMaxProvider")
+    public void testDecimalExceedingPrecisionMaxWithDecimalMapping(Integer typePrecision, Integer typeScale)
```

Comment:
Maybe instead of having single uber test that handles all the cases, maybe you could extract several tests for each of the case. With testing we should optimize the cost of reading the test code. This test is a bit complicated. WDYT?

Reply:
I will break it to 3 tests:
1) test with unsupported decimal values that must mapped to varchar
2) test with supported decimal values that must use rounding
3) test with supported decimal values that can be retrieved without rounding